godot-project/scenes/prefabs/Bowling/Ball/BowlingBall.gd

```diff
@@ -0,0 +1,5 @@
+extends RigidBody
+
+func _process(delta):
+	if global_transform.origin.y < 0:
+		queue_free()
```

Comment:
Maybe you could have used an area that detect the ball to delete it.

Reply:
I could have, I'm not sure it would have been any faster considering I'm only checking one axis.
Compared to an area that would check multiple, plus consider the shapes themselves and how they've intersected.
I've also applied a similar logic to the pins.